### DISCUSSION TITLE
Github Apps - Keep the state sent initially after admin installation approval?

### DISCUSSION THREAD

**Discussion by mathvdc** [community/community#125561] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

Hi, it's for sure related to [this topic ](https://github.com/orgs/community/discussions/61291)but I think now it's mostly can be considered as a bug.

When you initialize a new GithubApp bot installation with the `/new` endpoint to install it on an organization and it requires an admin approval -> in the callback this admin would have, we would find the `code` and the `installation_id` but not the "state" initially sent by the user asking the permission during the `/new `call.

It creates an annoying problem to link the `installation_id` to a specific user in our backend/database in the webhook handler.

Is it something solved today or is it still impossible to have this `state` in the callback? What other solution exists currently?

**Comment by ojaswa1942** [comment]
+1. This is something we regularly encounter in async installations.

**Reply by mathvdc** [reply]
yeah really annoying

**Comment by mathvdc** [comment]
anybody?

**Comment by zaibon** [comment]
This bug is also preventing me to have a proper installation flow for my application. Is there any workaround?

**Comment by brmzkw** [comment]
also interested :)

**Comment by pascal-botpress** [comment]
Encountering the same problem. We'd like to migrate from an OAuth App to a GitHub App, but without the state parameter, we're unable to do so

**Comment by lucasoares** [comment]
The `state` should be sent for all workflows when installing an app to an organization. It should also be sent when saving a configuration. Example:

```
- User goes to my app and starts the integration flow  
- For some reason, the app is already installed in their organization  
- User clicks on "Configure" on the installation page  
- User clicks on "Save" in the installation configuration  
- User is sent to the setup URL (Redirect on update option active on the app)  
- User is redirected to my app without the `state` parameter  
```

All workflows, webhooks, and everything directly related to the installation flow should send the `state` parameter so we can link our internal users and workspaces with the installation. This is currently a design flaw, as I have no way of knowing which user and workspace (a user on my app can have multiple workspaces) to link to the GitHub installation.

**Comment by kfsyamane** [comment]
+1
